Winston menu

winchan3 opened the pull request:
Menu use case
Search bar in the menu but no functionality (will be linked to search use case once merged)

winchan3 committed:
created menu files

winchan3 committed:
added few more menu files

winchan3 committed:
rudimentary outline logic in menuview file

winchan3 committed:
menustate basic structure and function

winchan3 committed:
layout for menuview

winchan3 committed:
small details added to files

winchan3 committed:
copied over viewmanagermodel (from cacoding) and finalized the usecase factory

also changed some dependencies so that each file would work with any class implementing a certain interface instead of directly depending on each other

winchan3 committed:
quick comment of some lines

now the program actually runs and i can see the menu screen. showed to teammates and they approved i guess.

winchan3 committed:
linked menu interactor and presenter

winchan3 committed:
javadoc for menu presenter and viewmodel

azyx0083 reviewed (approved):
Clearly follows the clean architecture. Existing JavaDocs are informative.
Good work overall, just some minor changes could be taken into consideration: 
Try creating a separate static method for MenuController in the MenuFactory because some other views also need a MenuContoller in their Factory.
MenuController doesn't need to execute a search. It only goes back to the menu page.
There is a TextButtonPanel from the CA-coding exercise. Try using it in your MenuView to have a better layout.

winchan3 committed:
added javadoc to all non-self-explanatory methods

also made a separate static method in menufactory to get the controller

winchan3 committed:
made menuview a little more presentable

set a predefined dimension for the menuview. also created proper center alignment of components and used filler objects to make it less cluttered
